### PR TITLE
Change application to match serialized response and show teaser

### PR DESCRIPTION
**Pull request by Mljungho**
**THIS PR CONTAINS**

- Updated response from server. Now considers a teaser text for articles.
- Updated test. All test passes.

Co-pilot @DefyCab 

Approval from: 
@elvitak @giacoletti 

https://www.pivotaltracker.com/story/show/181004474

### CODE REVIEW

File: src/components/SingleArticle.jsx [hunk 1]
```diff
@@ -17,8 +17,8 @@ const SingleArticle = () => {
   return (
     <>
       <h1 data-cy="article-title">{article.title}</h1>
-      <h1 data-cy="article-body">{article.body}</h1>
```
Comment (giacoletti) [inline src/components/SingleArticle.jsx:20]: In the "show" action of the article we get the full article body, we don't display a teaser.
From api:
{"article"=>
  {"id"=>206,
   "title"=>"Hello World",
   "body"=>"The world is a strange place",
   "category"=>"Mystring",
   "published"=>"2022-01-23T16:17:36.930Z"}}

File: cypress/integration/visitorCanVisitTheApplication.feature.js [hunk 2]
```diff
@@ -20,25 +20,18 @@ describe('Visitor can visit the application and', () => {
...
       cy.get('[data-cy=article-collection]').children().first().within(() => {
-        cy.get('[data-cy=article-body]').should('contain', 'Lorem ipsum');
-      });
-    });
-
-    it('is expected to see article creation date', () => {
-      cy.get('[data-cy=article-collection]').children().first().within(() => {
-        cy.get('[data-cy=article-created]')
-          .should('contain', '2022-01-13T10:24:38.354Z');
+        cy.get('[data-cy=article-created]').should('contain', '2022-01-23');
```
Comment (giacoletti) [inline cypress/integration/visitorCanVisitTheApplication.feature.js:34]: You can check if contains the full date: "2022-01-23T12:02:56.183Z"
No need to cut it to make the test pass.

File: cypress/integration/visitorCanVisitTheApplication.feature.js [hunk 3]
```diff
@@ -20,25 +20,18 @@ describe('Visitor can visit the application and', () => {
...
       cy.get('[data-cy=article-collection]').children().first().within(() => {
-        cy.get('[data-cy=article-title]').should('contain', 'Latin News');
+        cy.get('[data-cy=article-title]').should('contain', 'The Doors of Perception');
       });
     });
 
-    it('is expected to see article body', () => {
+      it('is expected to see article creation date', () => {
       cy.get('[data-cy=article-collection]').children().first().within(() => {
-        cy.get('[data-cy=article-body]').should('contain', 'Lorem ipsum');
```
Comment (giacoletti) [inline cypress/integration/visitorCanVisitTheApplication.feature.js:34]: You removed the spec to check the body, but you should have change it to check the teaser now

File: cypress/integration/visitorCanNavigateThroughCategories.feature.js [hunk 4]
```diff
@@ -22,7 +22,20 @@ describe("A user visiting the application can display articles by categories", (
     it("is expected to see a collection of articles", () => {
       cy.get("[data-cy=article-collection]")
         .children()
-        .should("have.length", 20);
+        .should("have.length", 4);
+    });
+
+    it("is expected to see teaser teaser", () => {
```
Comment (elvitak) [inline cypress/integration/visitorCanNavigateThroughCategories.feature.js]: This test is redundant. You are checking for the same thing in test that starts on line 52

File: cypress/integration/visitorCanSeeSingleArticle.feature.js [hunk 5]
```diff
@@ -16,14 +16,14 @@ describe("Visitor can see a single article", () => {
...
   });
 
   it("is expected to see article title", () => {
-    cy.get("[data-cy=article-title]").should("contain", "Latin News");
+    cy.get("[data-cy=article-title]").should("contain", "The Grapes of Wrath");
   });
 
   it("is expected to see article published date", () => {
-    cy.get("[data-cy=article-created]").should("contain", "2022-01-13");
+    cy.get("[data-cy=article-created]").should("contain", "2022-01-23");
```
Comment (giacoletti) [inline cypress/integration/visitorCanSeeSingleArticle.feature.js]: You can check if the full date is displayed, from the fixture

File: cypress/integration/visitorCanSeeSingleArticle.feature.js [hunk 6]
```diff
@@ -16,14 +16,14 @@ describe("Visitor can see a single article", () => {
   });
 
   it("is expected to display correct url", () => {
-    cy.url().should("contain", "360");
+    cy.url().should("contain", "1");
   });
 
   it("is expected to see article title", () => {
-    cy.get("[data-cy=article-title]").should("contain", "Latin News");
+    cy.get("[data-cy=article-title]").should("contain", "The Grapes of Wrath");
```
Comment (giacoletti) [inline cypress/integration/visitorCanSeeSingleArticle.feature.js:23]: spec for the article body missing

File: cypress/integration/visitorCanNavigateThroughCategories.feature.js [hunk 7]
```diff
@@ -190,10 +197,7 @@ describe("A user visiting the application can display articles by categories", (
         .children()
         .first()
         .within(() => {
-          cy.get("[data-cy=article-created]").should(
-            "contain",
-            "2022-01-13T10:24:38.354Z"
-          );
+          cy.get("[data-cy=article-created]").should("contain", "2022-01-23");
```
Comment (giacoletti) [inline cypress/integration/visitorCanNavigateThroughCategories.feature.js]: You can check for the full date

File: cypress/integration/visitorCanNavigateThroughCategories.feature.js (hunk 4 above)
Comment (giacoletti) [inline cypress/integration/visitorCanNavigateThroughCategories.feature.js]: "teaser teaser"?
Also this test is redundant, you do the exact same test below.

File: cypress/fixtures/sportsArticlesResponse.json [hunk 8]
```diff
@@ -3,37 +3,32 @@
     {
       "id": 482,
       "title": "Ronaldo and Messi",
-      "body": "Champions of whatever",
-      "created_at": "2021-12-13T10:24:38.354Z",
-      "updated_at": "2022-01-13T10:24:38.354Z"
+      "teaser": "Champions of whatever",
+      "published": "2021-12-13T10:24:38.354Z"
```
Comment (giacoletti) [inline cypress/fixtures/sportsArticlesResponse.json]: This fixture file doesn't match the api response.
We return the category as well.
Example:
{"id"=>236,
    "title"=>"Covid-19 is a pandemic",
    "teaser"=>"This is my article body...",
    "published"=>"2022-01-23T16:32:50.881Z",
    "category"=>"Politics"}

File: cypress/fixtures/politicsArticlesResponse.json [hunk 9]
```diff
@@ -3,142 +3,26 @@
     {
       "id": 360,
       "title": "Putin and Trump",
-      "body": "Shocking news, USA and Russia are...",
-      "created_at": "2022-01-13T10:24:38.354Z",
-      "updated_at": "2022-01-13T10:24:38.354Z"
+      "teaser": "Shocking news, USA and Russia are...",
+      "published": "2022-01-13T10:24:38.354Z"
```
Comment (giacoletti) [inline cypress/fixtures/politicsArticlesResponse.json]: Fixture doesn't match the api response:
{"id"=>236,
    "title"=>"Covid-19 is a pandemic",
    "teaser"=>"This is my article body...",
    "published"=>"2022-01-23T16:32:50.881Z",
    "category"=>"Politics"}
Category is missing.

File: cypress/fixtures/economyArticlesResponse.json [hunk 10]
```diff
@@ -3,37 +3,32 @@
     {
       "id": 482,
       "title": "Biggest financial crisis ever",
-      "body": "Tips and trics how not to become poor",
-      "created_at": "2021-12-10T10:24:38.354Z",
-      "updated_at": "2022-01-13T10:24:38.354Z"
+      "teaser": "Tips and trics how not to become poor",
+      "published": "2021-12-10T10:24:38.354Z"
```
Comment (giacoletti) [inline cypress/fixtures/economyArticlesResponse.json]: Category is missing, this fixture doesn't match the api response,
example:
{"id"=>236,
    "title"=>"Covid-19 is a pandemic",
    "teaser"=>"This is my article body...",
    "published"=>"2022-01-23T16:32:50.881Z",
    "category"=>"Politics"}

**Review by giacoletti — CHANGES_REQUESTED** [review]

File: cypress/integration/visitorCanSeeSingleArticle.feature.js (hunk 5 above)
Comment (Mljungho) [reply cypress/integration/visitorCanSeeSingleArticle.feature.js]: Ok

File: cypress/integration/visitorCanSeeSingleArticle.feature.js (hunk 6 above)
Comment (Mljungho) [reply cypress/integration/visitorCanSeeSingleArticle.feature.js:23]: added it block for body 

> 
> ```
>   it("is expected to see article body", () => {
>     cy.get("[data-cy=article-body]").should("contain", "Often people, especially computer engineers, focus on the machines. But in fact we need to focus on humans, on how humans care about doing programming or operating the application of the machines.");
>   });
> ```
>

File: cypress/integration/visitorCanNavigateThroughCategories.feature.js (hunk 7 above)
Comment (Mljungho) [reply cypress/integration/visitorCanNavigateThroughCategories.feature.js]: ok

File: cypress/fixtures/economyArticlesResponse.json (hunk 10 above)
Comment (Mljungho) [reply cypress/fixtures/economyArticlesResponse.json]: Fixed

**Pull request merged**